src/gwt/src/org/rstudio/studio/client/application/ui/CommandPalette.java

```diff
@@ -0,0 +1,442 @@
+/*
+ * CommandPalette.java
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+package org.rstudio.studio.client.application.ui;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.rstudio.core.client.DebouncedCommand;
+import org.rstudio.core.client.ElementIds;
+import org.rstudio.core.client.StringUtil;
+import org.rstudio.core.client.a11y.A11y;
+import org.rstudio.core.client.command.AppCommand;
+import org.rstudio.core.client.command.KeyMap;
+import org.rstudio.core.client.command.KeySequence;
+import org.rstudio.core.client.command.ShortcutManager;
+import org.rstudio.core.client.command.KeyMap.KeyMapType;
+import org.rstudio.core.client.js.JsUtil;
+import org.rstudio.studio.client.workbench.addins.Addins.AddinExecutor;
+import org.rstudio.studio.client.workbench.addins.Addins.RAddin;
+import org.rstudio.studio.client.workbench.addins.Addins.RAddins;
+import org.rstudio.studio.client.workbench.commands.Commands;
+
+import com.google.gwt.aria.client.ExpandedValue;
+import com.google.gwt.aria.client.Id;
+import com.google.gwt.aria.client.LiveValue;
+import com.google.gwt.aria.client.Roles;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.core.client.Scheduler;
+import com.google.gwt.dom.client.Element;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.resources.client.CssResource;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.FlowPanel;
+import com.google.gwt.user.client.ui.HTMLPanel;
+import com.google.gwt.user.client.ui.ScrollPanel;
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.user.client.ui.Widget;
+
+import elemental.events.KeyboardEvent.KeyCode;
+
+/**
+ * CommandPalette is a widget that displays all available RStudio commands in a
+ * searchable list.
+ */
+public class CommandPalette extends Composite
+{
+   private static CommandPaletteUiBinder uiBinder = GWT.create(CommandPaletteUiBinder.class);
+
+   interface CommandPaletteUiBinder extends UiBinder<Widget, CommandPalette>
+   {
+   }
+   
+   /**
+    * The host interface represents the class hosting the widget (not a widget
+    * itself), which is currently the CommandPaletteLauncher.
+    */
+   public interface Host
+   {
+      public void dismiss();
```

Comment:
IntelliJ nit: "public" is redundant on interface methods

Reply:
Updated.